Neos.Flow/Classes/Mvc/Controller/AbstractController.php

```diff
@@ -374,10 +383,23 @@ protected function throwStatus($statusCode, $statusMessage = null, $content = nu
      */
     protected function mapRequestArgumentsToControllerArguments()
     {
+        /** @var $argument Argument */
         foreach ($this->arguments as $argument) {
             $argumentName = $argument->getName();
             if ($this->request->hasArgument($argumentName)) {
                 $argument->setValue($this->request->getArgument($argumentName));
+
+                // Needed because we have an "auto-persistence" for mapped Resources since ages, but it was done in the
+                // TypeConverter previously, where validation could no longer prevent importing a resource into the system
+                $value = $argument->getValue();
+                if ($value instanceof PersistentResource && !$argument->getValidationResults()->hasErrors()) {
+                    try {
+                        $this->resourceManager->persistResource($value);
+                    } catch (\Exception $exception) {
+                        $this->logger->warning('Could not import an uploaded file', ['exception' => $exception] + LogEnvironment::fromMethodName(__METHOD__));
```

Comment:
Needed to keep the behaviour of import exceptions ending up as validation errors